### PR TITLE
Reduce surface of the python client in preparation of removing circular dependencies

### PR DESCRIPTION
**Pull request by EniasCailliau**
I also added comments referencing some of the Linear tickets

### CODE REVIEW

File: src/steamship/app/app.py [hunk 1]
```diff
@@ -72,7 +72,7 @@ def post(path: str, **kwargs):
     return endpoint(verb=Verb.POST, path=path, **kwargs)
 
 
-class App:
+class App:  # TODO (enias): Should be renamed to Invocable -> Solves redefinition of app
```
Comment (eob) [inline src/steamship/app/app.py:75]: Perhaps in a base class thought? 

From the user perspective, might make more sense to have `Package` and `Plugin` derive from `Invocable`

File: src/steamship/app/request.py [hunk 2]
```diff
@@ -36,7 +36,7 @@ class InvocationContext(CamelModel):
     invocable_type: str = None
 
 
-class Request(CamelModel):
+class Request(CamelModel):  # TODO (enias): Rename to ExecutableRequest?
```
Comment (eob) [inline src/steamship/app/request.py:39]: InvocableRequest? (given that that's the name you mentioned above in the PR)

**Review by eob — COMMENTED** [review]
LGTM so far; unsure if ready for merge or just placeholder for future PR!

File: src/steamship/app/request.py (hunk 2 above)
Comment (EniasCailliau) [reply src/steamship/app/request.py:39]: Yes agree! I'll feel it out today in another PR.

**Pull request merged**